### PR TITLE
Add UFW related rule for CIS on Ubuntu

### PR DESCRIPTION
**Pull request by dodys**
#### Description:

- This PR adds a few rules related to UFW (Uncomplicated Firewall) for CIS on Ubuntu:
  - ufw_rules_for_open_ports
  - set_ufw_loopback_traffic
  - set_ufw_default_rule

#### Rationale:

- Those rules are needed for CIS on Ubuntu 22.04 and/or 20.04.

### CODE REVIEW

**Comment by jan-cerny** [comment]
@dodys We're sorry but we had an issue with the CI jobs and unfortunately they're required to pass before merging. Could you please rebase this PR on the top of the latest upstream master branch? That will include fix from https://github.com/ComplianceAsCode/content/pull/10250 to this PR and will unblock the CI.

File: linux_os/guide/system/network/network-ufw/set_ufw_default_rule/rule.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+documentation_complete: true
+
+prodtype: ubuntu2004,ubuntu2204
+
+title: 'Ensure ufw default deny firewall policy'
```
Comment (marcusburghardt) [inline linux_os/guide/system/network/network-ufw/set_ufw_default_rule/rule.yml:5]: In alignment to [Style Guide](https://complianceascode.readthedocs.io/en/latest/manual/developer/04_style_guide.html#rule-sections) it must be in [Title case](https://en.wikipedia.org/wiki/Title_case)

File: linux_os/guide/system/network/network-ufw/set_ufw_default_rule/rule.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+description: |-
+    A default deny policy on connections ensures that any unconfigured
+    network usage will be rejected.
+
+    Note: Any port or protocol without a explicit allow before the default
+    deny will be blocked
```
Comment (marcusburghardt) [inline linux_os/guide/system/network/network-ufw/set_ufw_default_rule/rule.yml:12]:
```suggestion
    deny will be blocked.
```

File: linux_os/guide/system/network/network-ufw/set_ufw_default_rule/rule.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+rationale: |-
+    With a default accept policy the firewall will accept any packet that
+    is not configured to be denied. It is easier to white list acceptable
+    usage than to black list unacceptable usage.
```
Comment (marcusburghardt) [inline linux_os/guide/system/network/network-ufw/set_ufw_default_rule/rule.yml:17]:
```suggestion
    is not configured to be denied. It is easier to allow acceptable
    usage than to block unacceptable usage.
```

File: linux_os/guide/system/network/network-ufw/set_ufw_default_rule/sce/shared.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+#
```
Comment (marcusburghardt) [inline linux_os/guide/system/network/network-ufw/set_ufw_default_rule/sce/shared.sh:2]:
```suggestion
```

File: linux_os/guide/system/network/network-ufw/set_ufw_default_rule/sce/shared.sh [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+# platform = multi_platform_ubuntu
+# check-import = stdout
+#
```
Comment (marcusburghardt) [inline linux_os/guide/system/network/network-ufw/set_ufw_default_rule/sce/shared.sh:5]:
```suggestion
```

File: linux_os/guide/system/network/network-ufw/set_ufw_loopback_traffic/rule.yml [hunk 6]
```diff
@@ -0,0 +1,43 @@
+documentation_complete: true
+
+title: 'Set UFW loopback traffic'
```
Comment (marcusburghardt) [inline linux_os/guide/system/network/network-ufw/set_ufw_loopback_traffic/rule.yml:3]:
```suggestion
title: 'Set UFW Loopback Traffic'
```

File: linux_os/guide/system/network/network-ufw/set_ufw_loopback_traffic/rule.yml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    other interfaces should ignore traffic on this network as an
+    anti-spoofing measure.
+
+severity: medium
+
+references:
+    cis@ubuntu2004: 3.5.1.4
+    cis@ubuntu2204: 3.5.1.4
+
+warnings:
```
Comment (marcusburghardt) [inline linux_os/guide/system/network/network-ufw/set_ufw_loopback_traffic/rule.yml:23]: It should go after `ocil`

File: linux_os/guide/system/network/network-ufw/set_ufw_loopback_traffic/sce/shared.sh [hunk 8]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+#
```
Comment (marcusburghardt) [inline linux_os/guide/system/network/network-ufw/set_ufw_loopback_traffic/sce/shared.sh:2]:
```suggestion
```

File: linux_os/guide/system/network/network-ufw/ufw_rules_for_open_ports/sce/shared.sh [hunk 9]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+#
```
Comment (marcusburghardt) [inline linux_os/guide/system/network/network-ufw/ufw_rules_for_open_ports/sce/shared.sh:2]:
```suggestion
```

**Review by marcusburghardt — CHANGES_REQUESTED** [review]
My comments are more related to style guide.

File: linux_os/guide/system/network/network-ufw/set_ufw_loopback_traffic/rule.yml [hunk 10]
```diff
@@ -0,0 +1,43 @@
+documentation_complete: true
```
Comment (marcusburghardt) [inline linux_os/guide/system/network/network-ufw/set_ufw_loopback_traffic/rule.yml:1]:
```suggestion
documentation_complete: true

prodtype: ubuntu2004,ubuntu2204
```

Comment (marcusburghardt) [reply linux_os/guide/system/network/network-ufw/set_ufw_loopback_traffic/rule.yml:1]: Could you include `prodtype`, please? Then we limit this rule to not appear in other data streams.

**Review by marcusburghardt — APPROVED** [review]

**Comment by marcusburghardt** [comment]
Automatus CS8 and CS9 tests are failing because the rules are restricted to `prodtype: ubuntu2004,ubuntu2204`. Waived.

Overriding CODEOWNERS since a @dodys can't approve his own PRs.

**Pull request merged**